module/VuFind/src/VuFind/RecordDriver/Feature/HierarchyAwareTrait.php

```diff
@@ -100,7 +100,7 @@ public function setHierarchyDriverManager(
      */
     public function getHierarchyTopID()
     {
-        return $this->fields['hierarchy_top_id'] ?? [];
+        return (array)$this->fields['hierarchy_top_id'] ?? [];
```

Comment:
This should be `(array)($this->fields['hierarchy_top_id'] ?? [])` -- otherwise the typecasting will occur before the coalescing, and errors may result.

Reply:
Hmm, i did that but the rule 'no_unneeded_control_parentheses' => true, decided to say no. Ill try to adjust.

Reply:
Well, now it worked. Wonderous are the ways the code works sometimes.